### PR TITLE
Added ability to load prototypes recursively

### PR DESCRIPTION
**Pull request by edouardpoitras**
Not sure this PR is the cleanest way of doing it, but I needed that capability and it works on my end.
Please provide feedback and request changes as needed.

### CODE REVIEW

**Comment by edouardpoitras** [comment]
A few things maybe worth discussing:
1) Should we put a recursive depth limit?
2) I didn't bump the version - maybe worth going to 0.2.1?
3) A better/cleaner approach that I didn't think of?
4) Adding a parameter to with_dir/with_dirs instead of the two new variants with_dir_recursive/with_dirs_recursive? This means breaking backwards compatibility.

File: src/data.rs [hunk 1]
```diff
@@ -426,6 +456,23 @@ dyn_clone::clone_trait_object!(ProtoDeserializer);
...
+	/// # Examples
+	///
+	/// Example that recursively loads all yaml files from the assets/prototypes directory.
+	///
+	/// ```
+	/// use bevy_proto::ProtoDataOptions;
+	///
+	/// let opts = ProtoDataOptions {
+	/// 	directories: vec![String::from("assets/prototypes")],
+	///		recursive_loading: true,
```
Comment (MrGVSV) [inline src/data.rs:470]:
```suggestion
	///	recursive_loading: true,
```

File: src/plugin.rs [hunk 2]
```diff
@@ -25,6 +25,33 @@ impl ProtoPlugin {
...
 			options: Some(ProtoDataOptions {
 				directories: vec![dir.to_string()],
+				recursive_loading: false,
+				deserializer: Box::new(DefaultProtoDeserializer),
+				extensions: Some(vec!["yaml", "json"]),
+			}),
+		}
+	}
+
+	/// Same as with_dir but recursively loads prototype files.
```
Comment (MrGVSV) [inline src/plugin.rs:35]: Would be better to link this directly to the method so it's easier to find.

```suggestion
	/// Same as [with_dir] but recursively loads prototype files.
```

File: src/plugin.rs [hunk 3]
```diff
@@ -53,6 +80,36 @@ impl ProtoPlugin {
...
 			options: Some(ProtoDataOptions {
 				directories: dirs,
+				recursive_loading: false,
+				deserializer: Box::new(DefaultProtoDeserializer),
+				extensions: Some(vec!["yaml", "json"]),
+			}),
+		}
+	}
+
+	/// Same as with_dirs but recursively loads prototype files.
```
Comment (MrGVSV) [inline src/plugin.rs:90]:
```suggestion
	/// Same as [with_dirs] but recursively loads prototype files.
```

**Review by MrGVSV — CHANGES_REQUESTED** [review]
Looks good to me (just a few suggestions). Thanks for this!

**Comment by MrGVSV** [comment]
> 1. Should we put a recursive depth limit?

I don't think so. It could result in undefined behavior (unless we allow them to set the limit themselves, in which case, `recursive_loading` should probably be replaced with an `Option<u8>`).


> 2. I didn't bump the version - maybe worth going to 0.2.1?

Yeah, I'll bump and publish. Hoping to work on a version 0.3 release for Bevy 0.6 soon.


> 3. A better/cleaner approach that I didn't think of?

I find this approach pretty clean myself. If you want to add the recursion limit, that would be nice as well, but not super necessary at the moment.


> 4. Adding a parameter to with_dir/with_dirs instead of the two new variants with_dir_recursive/with_dirs_recursive? This would mean breaking backwards compatibility though.

I'm okay with the methods. I'd rather not add parameters to the others if we can help it (especially if those values are always typically going to be `true` or `false`, depending on the user's preference).

**Comment by edouardpoitras** [comment]
Great! Thanks.

**Review by MrGVSV — APPROVED** [review]
Awesome, thanks! I'll try to get a new version published sometime tonight or tomorrow.

**Pull request merged**